docs/architecture/adr-001-poap-contract.md

```diff
@@ -0,0 +1,177 @@
+* The `event_info` contains the event info.
+
+#### Cw721InstantiateMsg
+The following message instantiate the [CW721-base contract](https://github.com/CosmWasm/cw-nfts/tree/main/contracts/cw721-base):
+```rust
+pub struct Cw721InstatiateMsg {
+  pub name: String,
+  pub symbol: String,
+  pub minter: String,
+}
+```
+
+In the `POAP` contract case:
+* The `name` identifies the event name;
+* The `symbol` identifies the event logo (if exists);
+* The `minter` identifies the `POAP` contract address.
+
+##### EventInfo
+The `EventInfo` are used to instantiate the contract state with the information of the event.
+
+```rust
+pub struct EventInfo {
+    pub creator: Addr,
+    pub admin: Addr,
+    pub start_time: Timestamp,
+    pub end_time: Timestamp,
+    pub per_address_limit: u32,
+    pub base_poap_uri: String,
+    pub event_uri: String,
+    pub cw721_code_id: u64,
+}
+```
+
+* The `creator` field identifies the address of the event's creator;
+* The `admin` field identifies an optional admin for the contract, if not specified, the creator is the admin;
+* The `start_time` identifies the start time of the event;
+* The `end_time` identifies the end time of the event;
+* The `per_address_limit` identifies the max num of tokens that can be minted by an address;
+* The `base_poap_uri` identifies a valid `IPFS` URI corresponding to where the assets and metadata of the POAPs are stored.
+  * The `metadata` file is a `.json` that follow the `ERC-721` [metadata standard](https://docs.opensea.io/docs/metadata-standards#metadata-structure)
+* The field will be used to initialize the `token_uri` field of the `cw721-base` [`MintMsg<T>`](https://github.com/CosmWasm/cw-nfts/blob/1e992ccf640f07a384d6442625d6780a8e48ef1e/contracts/cw721-base/src/msg.rs#L61)
+* The `event_uri` field is used to initialize the `extension` part of the `cw721-base` [MintMsg< T >](https://github.com/CosmWasm/cw-nfts/blob/1e992ccf640f07a384d6442625d6780a8e48ef1e/contracts/cw721-base/src/msg.rs#L61) and contains a valid `IPFS` URI corresponding to where a `.json` file with the following event's metadata is stored:
+
+```json
+{
+  "name": "My awesome event",
+  "description": "Brief description of the event",
+  "city": "city where the event will be taking place",
+  "country": "country where the event will be taking place",
+  "start_date": "Start date for your event",
+  "end_date": "End date for your event",
+  "expiry_date": "From this day onward, no attendees will be able to mint POAPs from your event",
+  "year": <number>,
+  "event_url": "Event URL",
+  "virtual_event": true or false,
+}
+```
+
+#### Execute
+```rust
+pub enum ExecuteMsg {
+  EnableMint{},
+  Mint{},
+  MintTo{recipient: String},
+  UpdateTimes { start_time: Timestamp, end_time: Timestamp }
+}
+```
+
+#### EnableMint
+With the `EnableMint{}` message the creator or admin can enable the minting from any of the users.
+
+#### Mint
+With the `Mint{}` message a user can mint its own POAP.
```

Comment:
Allowing users to mint token would make those who don't attend the event be able to get POAP. Maybe we should remove this method.

Reply:
I believe that `EventManager` will solve this problem by `TryMint`. The event manager will be the admin of this contract. then call `MintTo` to get event token for the user. So, we can remove this method.